lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1544,6 +1544,10 @@ public void send(ProtocolMessage msg, boolean queueEvents, CompletionListener li
     private void sendImpl(ProtocolMessage message, CompletionListener listener) throws AblyException {
         if(transport == null) {
             Log.v(TAG, "sendImpl(): Discarding message; transport unavailable");
+            if (listener != null){
+                listener.onError(new ErrorInfo("Unable to send message; transport unavailable", 500, 50000));
```

Comment:
From what I know of our [error code ranges](https://github.com/ably/ably-common/tree/main/protocol#ranges) then the `5xx`/`5xxxx` range is intended to be used when we've had an HTTP `5xx` response back from an endpoint. Therefore, I wonder if `1xx`/`1xxxxx` was intended for use in this kind of scenario.

Reply:
I think 80009 makes sense so I updated it here b72945f1529331d3beafa8f1aff4d8a08793e96f . However I am not sure whether status and error codes be the same -> I haven't seen status codes for error codes above 50000

Reply:
Until we understand the circumstances in which this happens, how can we decide the error reason to return?

Reply:
This case was checking for `transport` and would return if it wasn't available. So I thought that was a suitable message/code. I didn't think about adding additional message / reason about the circumstances on why this is happening (in this case a concurrency issue). Would you have a recommendation on the error message code?